web/app.js

```diff
@@ -1369,8 +1369,13 @@ const PDFViewerApplication = {
       return;
     }
     const scripting = this.externalServices.scripting;
+    const {
+      info,
+      metadata,
+      contentDispositionFilename,
+    } = await pdfDocument.getMetadata();
 
-    window.addEventListener("updateFromSandbox", function (event) {
+    window.addEventListener("updateFromSandbox", event => {
```

Comment:
Generally speaking, given that the `case`s you're adding below *could* interfere with initial loading of the document and setting of the initial position, you should ensure that none of this code will run *until* the document has loaded at enough that it's possible to interact safely with it.
Hence, please add the following type of check here (unless we can delay the entire `this._initializeJavaScript(pdfDocument)` call to make this point moot):
```javascript
if (!this.isInitialViewSet) {
  return;
}
```